packages/slate/src/editor/marks.ts

```diff
@@ -4,21 +4,36 @@ import { Range } from '../interfaces/range'
 import { Path } from '../interfaces/path'
 import { Text } from '../interfaces/text'
 import { Element } from '../interfaces/element'
+import {Point} from "../interfaces";
+
+export const IS_FIREFOX =
+  typeof navigator !== 'undefined' &&
+  /^(?!.*Seamonkey)(?=.*Firefox).*/i.test(navigator.userAgent)
 
 export const marks: EditorInterface['marks'] = (editor, options = {}) => {
   const { marks, selection } = editor
 
   if (!selection) {
     return null
   }
+  let { anchor, focus } = selection
 
   if (marks) {
     return marks
   }
 
   if (Range.isExpanded(selection)) {
-    const [match] = Editor.nodes(editor, { match: Text.isText })
-
+    if (IS_FIREFOX) {
```

Comment:
I think we can safely remove this check in favour of cross-browser consistency

Reply:
isn't the issue only firefox-related ?

Reply:
i tried, it should work on both without the if, i removed it :D